container/libcontainer/handler.go

```diff
@@ -360,25 +368,20 @@ func schedulerStatsFromProcs(rootFs string, pids []int, pidMetricsCache map[int]
 
 // referencedBytesStat gets and clears referenced bytes
 // see: https://github.com/brendangregg/wss#wsspl-referenced-page-flag
-func referencedBytesStat(pids []int, cycles uint64, resetInterval uint64) (uint64, error) {
-	referencedKBytes, err := getReferencedKBytes(pids)
-	if err != nil {
-		return uint64(0), err
-	}
-
-	err = clearReferencedBytes(pids, cycles, resetInterval)
-	if err != nil {
-		return uint64(0), err
-	}
-	return referencedKBytes * 1024, nil
-}
 
 func getReferencedKBytes(pids []int) (uint64, error) {
 	referencedKBytes := uint64(0)
 	readSmapsContent := false
 	foundMatch := false
+	smapsFilePath := ""
 	for _, pid := range pids {
-		smapsFilePath := fmt.Sprintf(smapsFilePathPattern, pid)
+		smapsRollupFilePath := fmt.Sprintf(smaps_rollupFilePattern, pid)
```

Comment:
Yeah, but this is not a replacement, this is either or. If the rollup is not available, then it uses the normal one. I do not understand how I would be able to replace it smartly.

Reply:
I think that this should work :
```
smapsFilePath := ""
	for _, pid := range pids {
		smapsFilePath := fmt.Sprintf(smaps_rollupFilePattern, pid)
		if _, err := os.Stat(smapsFilePath ); err == nil {
			klog.V(6).Infof("Using smaps_rollup for pid %d instead of smaps", pid)
		} else {
			smapsFilePath = fmt.Sprintf(smapsFilePathPattern, pid)
		}
		smapsContent, err := ioutil.ReadFile(smapsFilePath)
```